### PR TITLE
Prepare wmde.9 (1.37)

### PR DESCRIPTION
**Pull request by addshore**
https://phabricator.wikimedia.org/T317890

Still todo...

- [x] Make this PR green
- [x] Remove the config options for feddy props https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/842868
- [x] Use the latest 1.37 MW image https://github.com/wmde/wikibase-release-pipeline/pull/357/commits/04f0970e6c3c119c33d7916a630f22c6d73e460b
- [x] Use newer mariadb images https://hub.docker.com/_/mariadb https://github.com/wmde/wikibase-release-pipeline/pull/357/commits/6743386970e6fc35731e8745b35b2f612a96b09c
- [x] Fix errors, Configure client entity sources https://github.com/wmde/wikibase-release-pipeline/pull/357/commits/48e778059fd0e88b6783bf01c0101ceeba818d89 https://github.com/wmde/wikibase-release-pipeline/pull/357/commits/8be1340c3910853d8cdaa635542aa729a7039d13
- [x] Fix repo RDF output with WikibaseLocalMedia fix https://github.com/ProfessionalWiki/WikibaseLocalMedia/pull/18 once merged
 - [x] Fix ConfirmEdit [Chrome Headless 93.0.4577.82 linux #0-0]    ✖ Should allow to edit with captcha eg. https://github.com/wmde/wikibase-release-pipeline/actions/runs/3169163570/jobs/5160983595
 - [x] Make sure we use the tagged release https://github.com/ProfessionalWiki/WikibaseLocalMedia/releases/tag/1.0.2
 - [x] Fix repo_client/item.js `Should be able to see delete changes is dispatched to client for test page` eg. https://github.com/wmde/wikibase-release-pipeline/actions/runs/3169163570/jobs/5160983359#step:4:2641 & Fix repo_client/extensions/scribunto-item.js `Should be able to see delete changes is dispatched to client for lua page` eg. https://github.com/wmde/wikibase-release-pipeline/actions/runs/3169163570/jobs/5160983359#step:4:2669
   -Requires multiple backports of reverts to the Wikibase extension

### CODE REVIEW

**Comment by addshore** [comment]
Only 1.37.3 is currently built into an official docker image, hence the current failure

File: CHANGES.md [hunk 1]
```diff
@@ -4,12 +4,27 @@ This file is only intended to serve the developers of this repository and of the
...
+- [MediaWiki release notes](https://www.mediawiki.org/wiki/Release_notes/1.37)
+- [Wikibase release notes](https://github.com/wikimedia/Wikibase/blob/REL1_37/RELEASE-NOTES-1.37)
+
+The Query service was also updated from `0.3.97` to TODO, but no additional upgrade process is needed.
 
 | Suite Version | MediaWiki release | Date availbile | Run number | Release task  |
 |---------------|----------------------|----------------|------------| --------------|
-| wmde.8 spec(TBA) | 1.36    | Date TBA | Run TBA | [T316707](https://phabricator.wikimedia.org/T316707) |
-| wmde.7 spec(TBA) | 1.35    | Date TBA | Run TBA | [T314881](https://phabricator.wikimedia.org/T314881) |
+| wmde.9 ([spec](https://github.com/wmde/wikibase-release-pipeline/blob/wmde.9/versions/wmde9.env),[metadata](https://github.com/wmde/wikibase-release-pipeline/blob/main/versions/wmde9)) | 1.37.TODO    | TODO | [TODO](https://github.com/wmde/wikibase-release-pipeline/actions/runs/TODO) | [T317890](https://phabricator.wikimedia.org/T317890) |
```
Comment (JeroenDeDauw) [inline CHANGES.md:18]: Would be good to list both MW and PHP versions, not just MW

**Comment by addshore** [comment]
Awaiting https://gerrit.wikimedia.org/r/c/wikidata/query/gui/+/836231 Switch grunt-merge-i18n back to https in npm (#2) [NEW] which will then need to be used for this release.

File: CHANGES.md (hunk 1 above)
Comment (addshore) [reply CHANGES.md]: Done in https://github.com/wmde/wikibase-release-pipeline/pull/357/commits/59cf15219809b34e3e636c43158c70863d75c0f0

**Comment by addshore** [comment]
mariadb_10_3_base__repo_client seems to be having some fun...

image

It looks like the exception stacktrace doesn't end up in any of the logs, so I'll need to modify CI to spit it out somehow

**Comment by addshore** [comment]
Special%3ANewItem-should-not-be-accessible-on-client-2022-09-29T18-09-26-033Z (1)

**Comment by addshore** [comment]
New 1.37 releases were made, so I also prepared the mediawiki docker images for these
https://github.com/wikimedia/mediawiki-docker/pull/114
https://github.com/docker-library/official-images/pull/13257
Once both merged they should be used here.

**Comment by addshore** [comment]
Special%3ANewItem-should-not-be-accessible-on-client-2022-09-30T09-14-53-646Z (1)

**Comment by addshore** [comment]
Next failure appears to be `[Chrome Headless 93.0.4577.82 linux #0-1]    ✖ Should be able to see delete changes is dispatched to client for test page` for the client tests.
I guess this is due to the changes to how dispatching works that are half complete in the current REL1_37 branch.

**Comment by addshore** [comment]
Current failing tests just result in a blank page, how useful!
Should-be-able-to-insert-interwiki-links-2022-09-30T11-19-41-080Z

**Comment by addshore** [comment]
Right, mw log output in CI works, just the `tail` needs to only be run if there are files / not error if there are none
image

**Comment by addshore** [comment]
One of the issues appears to be with WikibaseLocalMedia extension causing RDF retrieval via Special:EntityData to fail.

```
[365f46590252b4828c500610] /wiki/Special:EntityData/Q17.ttl?flavor=dump&nocache=1664726210237 Error: Undefined class constant 'OBJECT_PROPERTY'

Backtrace:

from /var/www/html/extensions/WikibaseLocalMedia/src/HookHandlers.php(29)
#0 /var/www/html/extensions/Wikibase/lib/includes/DataTypeDefinitions.php(336): Wikibase\LocalMedia\HookHandlers::Wikibase\LocalMedia\{closure}()
#1 [internal function]: Wikibase\Lib\DataTypeDefinitions->Wikibase\Lib\{closure}(Closure)
#2 /var/www/html/extensions/Wikibase/lib/includes/DataTypeDefinitions.php(342): array_map(Closure, array)
#3 /var/www/html/extensions/Wikibase/repo/WikibaseRepo.entitytypes.php(388): Wikibase\Lib\DataTypeDefinitions->getRdfDataTypes()
#4 [internal function]: Wikimedia\Services\ServiceContainer->{closure}(integer, Wikibase\Repo\Rdf\RdfVocabulary, Wikimedia\Purtle\TurtleRdfWriter, Wikibase\Repo\Rdf\RdfBuilder, Wikibase\Repo\Rdf\HashDedupeBag)
#5 /var/www/html/extensions/Wikibase/repo/includes/Rdf/EntityRdfBuilderFactory.php(87): call_user_func(Closure, integer, Wikibase\Repo\Rdf\RdfVocabulary, Wikimedia\Purtle\TurtleRdfWriter, Wikibase\Repo\Rdf\RdfBuilder, Wikibase\Repo\Rdf\HashDedupeBag)
#6 /var/www/html/extensions/Wikibase/repo/includes/Rdf/EntityRdfBuilderFactory.php(59): Wikibase\Repo\Rdf\EntityRdfBuilderFactory->createEntityRdfBuilders(integer, Wikibase\Repo\Rdf\RdfVocabulary, Wikimedia\Purtle\TurtleRdfWriter, Wikibase\Repo\Rdf\RdfBuilder, Wikibase\Repo\Rdf\HashDedupeBag)
#7 /var/www/html/extensions/Wikibase/repo/includes/Rdf/RdfBuilder.php(101): Wikibase\Repo\Rdf\EntityRdfBuilderFactory->getEntityRdfBuilders(integer, Wikibase\Repo\Rdf\RdfVocabulary, Wikimedia\Purtle\TurtleRdfWriter, Wikibase\Repo\Rdf\RdfBuilder, Wikibase\Repo\Rdf\HashDedupeBag)
#8 /var/www/html/extensions/Wikibase/repo/includes/Rdf/RdfBuilderFactory.php(45): Wikibase\Repo\Rdf\RdfBuilder->__construct(Wikibase\Repo\Rdf\RdfVocabulary, Wikibase\Repo\Rdf\EntityRdfBuilderFactory, integer, Wikimedia\Purtle\TurtleRdfWriter, Wikibase\Repo\Rdf\HashDedupeBag, Wikibase\Repo\Content\EntityContentFactory, Wikibase\Repo\Rdf\EntityStubRdfBuilderFactory, Wikibase\Lib\Store\CachingEntityRevisionLookup)
#9 /var/www/html/extensions/Wikibase/repo/includes/LinkedData/EntityDataSerializationService.php(338): Wikibase\Repo\Rdf\RdfBuilderFactory->getRdfBuilder(integer, Wikibase\Repo\Rdf\HashDedupeBag, Wikimedia\Purtle\TurtleRdfWriter)
#10 /var/www/html/extensions/Wikibase/repo/includes/LinkedData/EntityDataSerializationService.php(146): Wikibase\Repo\LinkedData\EntityDataSerializationService->createRdfBuilder(string, string)
#11 /var/www/html/extensions/Wikibase/repo/includes/LinkedData/EntityDataRequestHandler.php(551): Wikibase\Repo\LinkedData\EntityDataSerializationService->getSerializedData(string, Wikibase\Lib\Store\EntityRevision, NULL, array, string)
#12 /var/www/html/extensions/Wikibase/repo/includes/LinkedData/EntityDataRequestHandler.php(285): Wikibase\Repo\LinkedData\EntityDataRequestHandler->showData(WebRequest, OutputPage, string, Wikibase\DataModel\Entity\ItemId, integer)
#13 /var/www/html/extensions/Wikibase/repo/includes/Specials/SpecialEntityData.php(112): Wikibase\Repo\LinkedData\EntityDataRequestHandler->handleRequest(string, WebRequest, OutputPage)
#14 /var/www/html/includes/specialpage/SpecialPage.php(647): Wikibase\Repo\Specials\SpecialEntityData->execute(string)
#15 /var/www/html/includes/specialpage/SpecialPageFactory.php(1366): SpecialPage->run(string)
#16 /var/www/html/includes/MediaWiki.php(314): MediaWiki\SpecialPage\SpecialPageFactory->executePath(string, RequestContext)
#17 /var/www/html/includes/MediaWiki.php(932): MediaWiki->performRequest()
#18 /var/www/html/includes/MediaWiki.php(563): MediaWiki->main()
#19 /var/www/html/index.php(53): MediaWiki->run()
#20 /var/www/html/index.php(46): wfIndexMain()
#21 {main}
```

the class constant is used here https://github.com/ProfessionalWiki/WikibaseLocalMedia/blob/def6564ba847181b60b2958efa4508d1512e9d55/src/HookHandlers.php#L24-L30
However it doesn't exist in REL1_37 https://github.com/wikimedia/Wikibase/blob/REL1_37/repo/includes/Rdf/PropertyRdfBuilder.php
I did exist in REL1_36 https://github.com/wikimedia/Wikibase/blob/REL1_36/repo/includes/Rdf/PropertyRdfBuilder.php#L17

It looks like this constant is now in `PropertySpecificComponentsRdfBuilder`
https://github.com/wikimedia/Wikibase/blob/REL1_37/repo/includes/Rdf/PropertySpecificComponentsRdfBuilder.php#L25

PR inbound for the extension fixing this issue.

**Comment by addshore** [comment]
I verified fixes for the tests in https://github.com/wmde/wikibase-release-pipeline/pull/374
For Wikibase this is primarily made up of:
 - 841860: Revert "Avoid inserting wb_changes entry when there is no subscriber" | https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/841860
 - 841861: Revert "Introduce ChangeHolder and use that to store the change later" | https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/841861
 - 841862: Revert "Use UserIndetity directly instead of User object" | https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/841862
 
In order to make REL1_37 CI green I also had to merge 2 other things...
 - 842800: Allow dealerdirect/phpcodesniffer-composer-installer plugin | https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/842800
 - 842853: Use assertEqualsWithDelta to fix RescoreBuilderTest | https://gerrit.wikimedia.org/r/c/mediawiki/extensions/CirrusSearch/+/842853
 
 When this is all merged, a rebuild here should result in a green PR

**Comment by addshore** [comment]
The 3 reverts on Wikibase are merged, so the next build should be green / free of the errors relating to dispatching

**Comment by addshore** [comment]
https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/842868 is green on gerrit now...
Next steps are:
 - Merge it
 - Rebuild this PR
 - PR should be green
 - Continue release checklist on https://phabricator.wikimedia.org/T317890

**Comment by addshore** [comment]
This should probably not be squashed, as there is some value in the split commits I think

**Comment by addshore** [comment]
> It is a pile of fairly random commits though. I understand it was CI-driven development - any chances separate conceptual changes (e.g. steps in the list in the description of this PR) could be grouped somehow? They seem to be all over the shop currently?

I can give that a go

**Comment by addshore** [comment]
Infact, looking at the commits, and the fact that the wmde.8 and .7 PRs were squashed, lets just squash this one!

**Pull request merged**